R/geotargets-package.R

```diff
@@ -7,6 +7,7 @@ NULL
 
 # clear CRAN checks spotting floating global variables
 #' @importFrom utils globalVariables
+#' @importFrom rlang %||% arg_match0 is_installed check_installed
```

Comment:
importing `arg_match0`, `is_installed` and `check_installed` with `@importFrom` is only needed if we want to use those functions without namespacing them. My preference is that we don't @importFrom these functions as they aren't used very often.

It is necessary to importFrom `%||%` at the moment, although I note that in R 4.4.0 `%||%` is implemented natively, which is very cool.

Reply:
I have removed all three from importFrom as requested.. Indeed, only using `is_installed()` for `@examplesIf` block to check for stars, which may not really be necessary but I think we don't want to assume suggested packages are installed, in general. `check_installed()` is not used but I initially thought we would use it, thanks for pointing that out. `arg_match0` is used in a couple places now for terra and stars drivers checks.

For imports I generally prefer to handle functions explicitly, as well as use double colon to access them; the latter because it is easier for humans to see what is coming from where, and the former to explicitly declare what is used by the package in the NAMESPACE--no matter how commonly it is used.

Reply:
Thank you! :)

> For imports I generally prefer to handle functions explicitly, as well as use double colon to access them; the latter because it is easier for humans to see what is coming from where, and the former to explicitly declare what is used by the package in the NAMESPACE--no matter how commonly it is used.

I hear you! I prefer to use `pkg::fun()`, with the only exception being for imported infix operators, e.g., `%||%`. Another compelling case I've seen is for packages that make heavy use of ggplot2 functions or rlang metaprogramming, then you can use `@import ggplot2` etc.

 > explicitly declare what is used by the package in the NAMESPACE--no matter how commonly it is used.

I think this is a good idea, and it can be helpful to identify which functions are actually used from a package. It does have the potential to get a bit gnarly if you are making heavy use of importing from various packages though. I haven't seen much more development of the [itdepends](https://github.com/r-lib/itdepends) pkg, but I think tools like that would be really useful. The reason I don't use @importFrom very often is because of reasons stated in the [rpkgs book section on dependencies](https://r-pkgs.org/dependencies-in-practice.html#sec-dependencies-in-imports-r-code):

> @importFrom aaapkg aaa_fun. This does suppress the NOTE, but it also does more. This causes aaapkg to be loaded whenever your package is loaded. In contrast, if you use the approach we recommend, the aaapkg will only be loaded if your user does something that actually requires it. This rarely matters in practice, but it’s always nice to minimize or delay the loading of additional packages.

Anyway, that is a bit of an aside! I want to say thanks again for all your continued help with this project. I'm now curious if the `itdepends` project can be revived 🤔